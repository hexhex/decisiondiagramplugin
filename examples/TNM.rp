[common signature]
predicate: root/1;
predicate: innernode/1;
predicate: leafnode/2;
predicate: conditionaledge/5;
predicate: elseedge/2;

[belief base]
name: tnm;
mapping: "
	root(checkM).

	innernode(checkM).
	innernode(checkT).
	innernode(checkN).

	leafnode(stage1, stage1).
	leafnode(stage2, stage2).
	leafnode(stage3, stage3).
	leafnode(stage4, stage4).

	conditionaledge(checkM, stage4, m, \'=\', 1).
	elseedge(checkM, checkN).

	conditionaledge(checkN, stage3, n, \'>\', 0).
	elseedge(checkN, checkT).

	conditionaledge(checkT, stage2, t, \'>=\', 3).
	elseedge(checkT, stage1).
";

[revision plan]
{

				tnm
}
