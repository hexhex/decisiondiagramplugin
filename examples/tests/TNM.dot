digraph {
     checkN -> checkT [label="else"];
     checkT -> stage1 [label="else"];
     checkM -> checkN [label="else"];
     checkN -> stage3 [label="n>0"];
     checkT -> stage2 [label="t>=3"];
     checkM -> stage4 [label="m=1"];
     stage4 [label="stage4 [stage2]"];
     stage3 [label="stage3 [stage2]"];
     stage2 [label="stage2 [stage2]"];
     stage1 [label="stage1 [stage1]"];
     checkN [label="checkN"];
     checkT [label="checkT"];
     checkM [label="checkM"];
}
