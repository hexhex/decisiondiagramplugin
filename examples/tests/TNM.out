{elseedge("check n","check t"), elseedge("check t",stage1), elseedge("check m","check n"), conditionaledge("check n",stage3,n,">",0), conditionaledge("check t",stage2,t,">=",3), conditionaledge("check m",stage4,m,"=",1), leafnode(stage4,stage2), leafnode(stage3,stage2), leafnode(stage2,stage2), leafnode(stage1,stage1), innernode("check n"), innernode("check t"), innernode("check m"), root("check m")}
